br/pkg/task/restore.go

```diff
@@ -1117,6 +1117,10 @@ func enableTiDBConfig() func() {
 		// when upstream and downstream both set this value greater than default(3072)
 		conf.MaxIndexLength = config.DefMaxOfMaxIndexLength
```

Comment:
seems fail on same type of mistake again? Do you consider to skip checks when creating table? like https://github.com/pingcap/tidb/blob/522a00840385dfa43509a3af2bc79e6ff50252a8/ddl/schematracker/dm_tracker.go#L178

Reply:
https://github.com/pingcap/tidb/blob/522a00840385dfa43509a3af2bc79e6ff50252a8/ddl/schematracker/dm_tracker_test.go#L50

DM has met a similar problem before, which is upstream MySQL has more capability than TiDB, so we use a special DDL implementation to apply DDL SQL.

Reply:
There is a check before create table with DDL.
https://github.com/pingcap/tidb/blob/57d778f0b1b6daebe0f330af2baa54e9e5310ea4/ddl/ddl_api.go#L2337-L2352

When use br, every check config is default value. If upstream has large value, for example set `TableColumnCountLimit` to 4000, the restore might be failed when there is a table with 4000 columns.

Reply:
BR import the DDL package, but doesn't expose these configs.